### DISCUSSION TITLE
SQLAlchemy version 1.4.42 is unable to fetch_all from mysql db.

### DISCUSSION THREAD

**Discussion by sadfiesch** [sqlalchemy/sqlalchemy#8698] [Usage Questions]
### Describe the bug

When following the FastAPI Async Rational DB(https://fastapi.tiangolo.com/advanced/async-sql-databases/) turtorial I was unable to complete it as I kept getting the issue(see stackoverflow https://stackoverflow.com/questions/74089620/python-databases-library-cant-fetch-all-from-mysql-database) 
However after downgrading it fixed the issue

### To Reproduce

```python
see fastapi turtorial
```

### Error

  File "c:\users\nikhi\desktop\upwork\srinath srinath\api_project\venv\lib\site-packages\databases\backends\mysql.py", line 115, in fetch_all
    metadata = CursorResultMetaData(context, cursor.description)
  File "c:\users\nikhi\desktop\upwork\srinath srinath\api_project\venv\lib\site-packages\sqlalchemy\engine\cursor.py", line 164, in __init__
    (
ValueError: not enough values to unpack (expected 5, got 4)

### Versions

- OS:
- Python: 3,10
- SQLAlchemy: 1.4.42
- Database: Mysql seems to be every version
- DBAPI (eg: psycopg, cx_oracle, mysqlclient):

**Comment by CaselIT** [comment]
Hi,

This seem something in an external library. Please report the issue with them. Without a reproduction example that includes only sqlalchemy code we can't do much

**Reply by sadfiesch** [reply]
Since downgrading sqlalchemy solved the issue i doubt the issue is somewhere else

**Reply by zzzeek** [reply]
I can guarantee you the issue is somewhere else, the two lines in your stack trace say everything:

    databases\backends\mysql.py

this is not SQLAlchemy.  This is a package called [databases](https://github.com/encode/databases).  It is based on [deep, extensive access to SQLAlchemy's private internals](https://github.com/encode/databases/blob/385f3fd788f7a513ca1ee6a8efe5316b3d93f6cd/databases/backends/mysql.py#L115) which is a fully incorrect way to write Python code.  They are then passing it a [random object which they invented themselves](https://github.com/encode/databases/blob/master/databases/backends/mysql.py#L210) that just happens to look like one of SQLAlchemy's [own objects](https://github.com/sqlalchemy/sqlalchemy/blob/8aa07c8e62a0483e0c882ffbf97453d5e4614efe/lib/sqlalchemy/engine/default.py#L974) but unfortunately not enough, because [we can change these internals at any time](https://github.com/sqlalchemy/sqlalchemy/commit/bbd90a987eea70c28dd1e2ee7007722b16ec9f74).

I would also suggest please reading (and following!) the [bug reporting guidelines](https://github.com/sqlalchemy/sqlalchemy/blob/main/.github/ISSUE_TEMPLATE/bug_report.yaml) which should have displayed in the issue template before reporting further issues.  Thanks!

**Reply by CaselIT** [reply]
> Since downgrading sqlalchemy solved the issue i doubt the issue is somewhere else

Well you seem to be using an extenal library that internally use sqlalchemy. The fact that one version of sqlalchemy works and the other don't suggests that the issue is in the external library. Sqlalchemy developer have no control over external libraries.
